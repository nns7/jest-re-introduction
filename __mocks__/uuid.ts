const v4 = () => "00000000-0000-0000-0000-000000000000";
export { v4 };
