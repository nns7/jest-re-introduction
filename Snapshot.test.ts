import { v4 as uuidv4 } from "uuid";

describe("Snapshot Testing", () => {
  test("toMatchSnapshot - 基本", () => {
    const html = `
    <div class="container">
      <article>
        <p class="title">UI生成結果<p>
      </article>
    </div>
    `;
    expect(html).toMatchSnapshot();
  });

  test("toMatchSnapshot - インライン", () => {
    const html = `
    <div class="container">
      <article>
        <p>UI生成結果<p>
      </article>
    </div>
    `;
    expect(html).toMatchInlineSnapshot(`
"
    <div class="container">
      <article>
        <p>UI生成結果<p>
      </article>
    </div>
    "
`);
  });

  test("toMatchSnapshot - Property Matchers", () => {
    const obj = {
      id: uuidv4(),
      created: new Date().getTime(),
      type: "jest",
    };
    expect(obj).toMatchSnapshot({
      id: expect.stringMatching(
        /^[\da-f]{8}-[\da-f]{4}-[\da-f]{4}-[\da-f]{4}-[\da-f]{12}$/
      ),
      created: expect.any(Number),
    });
  });
});
