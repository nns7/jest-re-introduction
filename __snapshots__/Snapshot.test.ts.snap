// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing toMatchSnapshot - Property Matchers 1`] = `
{
  "created": Any<Number>,
  "id": StringMatching /\\^\\[\\\\da-f\\]\\{8\\}-\\[\\\\da-f\\]\\{4\\}-\\[\\\\da-f\\]\\{4\\}-\\[\\\\da-f\\]\\{4\\}-\\[\\\\da-f\\]\\{12\\}\\$/,
  "type": "jest",
}
`;

exports[`Snapshot Testing toMatchSnapshot - インライン 1`] = `
"
    <div class="container">
      <article>
        <p>UI生成結果<p>
      </article>
    </div>
    "
`;

exports[`Snapshot Testing toMatchSnapshot - 基本 1`] = `
"
    <div class="container">
      <article>
        <p class="title">UI生成結果<p>
      </article>
    </div>
    "
`;
